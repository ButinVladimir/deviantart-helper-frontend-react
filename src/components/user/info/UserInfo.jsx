import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import Content from 'react-bulma-components/lib/components/content';
import Button from 'react-bulma-components/lib/components/button';
import Heading from 'react-bulma-components/lib/components/heading';
import Config from '../../../config/config';

export default class UserInfo extends Component {
  componentDidMount() {
    const { fetchUserData } = this.props;

    fetchUserData();
  }

  render() {
    const {
      userInfoLoading,
      userName,
      userId,
      userType,
      accessTokenExpires,
      refreshTokenExpires,
      refreshHandler,
    } = this.props;

    return (
      <Section>
        <Container>
          <Content>
            <Heading size={2}>Current user info</Heading>
            <p>
              <span>User: </span>
              <strong>{userName}</strong>
            </p>
            <p>
              <span>User id: </span>
              <strong>{userId}</strong>
            </p>
            <p>
              <span>User type: </span>
              <strong>{userType}</strong>
            </p>
            <p>
              <span>Access token expires: </span>
              <strong>{new Date(accessTokenExpires).toLocaleString()}</strong>
            </p>
            <p>
              <span>Refresh token expires: </span>
              <strong>{new Date(refreshTokenExpires).toLocaleString()}</strong>
            </p>
            <p>
              <Button color="primary" loading={userInfoLoading} onClick={refreshHandler}>Refresh tokens</Button>
            </p>
          </Content>
        </Container>
      </Section>
    );
  }
}

UserInfo.propTypes = {
  // eslint-disable-next-line react/no-unused-prop-types
  config: PropTypes.instanceOf(Config).isRequired,
  fetchUserData: PropTypes.func.isRequired,
  userInfoLoading: PropTypes.bool.isRequired,
  userId: PropTypes.string,
  userName: PropTypes.string,
  userType: PropTypes.string,
  accessTokenExpires: PropTypes.number,
  refreshTokenExpires: PropTypes.number,
  refreshHandler: PropTypes.func.isRequired,
};

UserInfo.defaultProps = {
  userId: '',
  userName: '',
  userType: '',
  accessTokenExpires: 0,
  refreshTokenExpires: 0,
};
