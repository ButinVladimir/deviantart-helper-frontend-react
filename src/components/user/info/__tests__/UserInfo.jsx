import React from 'react';
import { shallow } from 'enzyme';
import UserInfo from '../UserInfo';

describe('UserInfo', () => {
  it('can be rendered correctly using default props', () => {
    const wrapper = shallow(<UserInfo fetchUserData={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('can be rendered correctly', () => {
    const wrapper = shallow(
      <UserInfo
        fetchUserData={jest.fn()}
        userId="123-456"
        userName="UserName"
        userType="regular"
        accessTokenExpires={1001}
        refreshTokenExpires={2001}
        fetchDateThreshold={3001}
        requestDateThreshold={4001}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('handles mounting', () => {
    const fetchUserData = jest.fn();
    shallow(<UserInfo fetchUserData={fetchUserData} />);

    expect(fetchUserData).toHaveBeenCalled();
  });
});
