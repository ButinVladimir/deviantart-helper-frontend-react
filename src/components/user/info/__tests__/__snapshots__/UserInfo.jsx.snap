// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo can be rendered correctly 1`] = `
<Section
  className=""
  renderAs="section"
  size={null}
  style={Object {}}
>
  <Container
    breakpoint={null}
    fluid={false}
    renderAs="div"
    style={Object {}}
  >
    <Content
      renderAs="div"
      size={null}
      style={Object {}}
    >
      <Heading
        className=""
        heading={false}
        renderAs="h1"
        size={2}
        spaced={false}
        subtitle={false}
        weight={null}
      >
        Current user info
      </Heading>
      <p>
        <span>
          User: 
        </span>
        <strong>
          UserName
        </strong>
      </p>
      <p>
        <span>
          User id: 
        </span>
        <strong>
          123-456
        </strong>
      </p>
      <p>
        <span>
          User type: 
        </span>
        <strong>
          regular
        </strong>
      </p>
      <p>
        <span>
          Access token expires: 
        </span>
        <strong>
          1970-1-1 03:00:01
        </strong>
      </p>
      <p>
        <span>
          Refresh token expires: 
        </span>
        <strong>
          1970-1-1 03:00:02
        </strong>
      </p>
      <p>
        <span>
          Fetch date threshold: 
        </span>
        <strong>
          1970-1-1 03:00:03
        </strong>
      </p>
      <p>
        <span>
          Request date threshold: 
        </span>
        <strong>
          1970-1-1 03:00:04
        </strong>
      </p>
    </Content>
  </Container>
</Section>
`;

exports[`UserInfo can be rendered correctly using default props 1`] = `
<Section
  className=""
  renderAs="section"
  size={null}
  style={Object {}}
>
  <Container
    breakpoint={null}
    fluid={false}
    renderAs="div"
    style={Object {}}
  >
    <Content
      renderAs="div"
      size={null}
      style={Object {}}
    >
      <Heading
        className=""
        heading={false}
        renderAs="h1"
        size={2}
        spaced={false}
        subtitle={false}
        weight={null}
      >
        Current user info
      </Heading>
      <p>
        <span>
          User: 
        </span>
        <strong />
      </p>
      <p>
        <span>
          User id: 
        </span>
        <strong />
      </p>
      <p>
        <span>
          User type: 
        </span>
        <strong />
      </p>
      <p>
        <span>
          Access token expires: 
        </span>
        <strong>
          Null
        </strong>
      </p>
      <p>
        <span>
          Refresh token expires: 
        </span>
        <strong>
          Null
        </strong>
      </p>
      <p>
        <span>
          Fetch date threshold: 
        </span>
        <strong>
          Null
        </strong>
      </p>
      <p>
        <span>
          Request date threshold: 
        </span>
        <strong>
          Null
        </strong>
      </p>
    </Content>
  </Container>
</Section>
`;
